code/game/gamemodes/abduction/abduction.dm

```diff
@@ -0,0 +1,342 @@
+/datum/game_mode
+	var/list/datum/mind/abductors = list()
+
+/datum/game_mode/abduction
+	name = "Abduction"
+	config_tag = "abduction"
+	antag_flag = BE_ABDUCTOR
+	pre_setup_before_jobs = 1
+	recommended_enemies = 2
+	required_players = 15
+	var/max_teams = 4
+	var/teams = 1
+	var/list/datum/mind/scientists = list()
+	var/list/datum/mind/agents = list()
+	var/list/datum/objective/team_objectives = list()
+	var/list/team_names = list()
+
+/datum/game_mode/abduction/announce()
+	world << "<B>The current game mode is - Abduction!</B>"
+
+/datum/game_mode/abduction/pre_setup()
+	var/teams = max(1, min(max_teams,round(num_players()/config.abductor_scaling_coeff)))
+	var/possible_teams = max(1,round(antag_candidates.len / 2))
+	teams = min(teams,possible_teams)
+
+	abductors.len = 2*teams
+	scientists.len = teams
+	agents.len = teams
+	team_objectives.len = teams
+	team_names.len = teams
+
+	for(var/i=1,i<=teams,i++)
+		if(!make_abductor_team(i))
+			return 0
+
+	return 1
+
+/datum/game_mode/abduction/proc/make_abductor_team(var/team_number)
+	//Team Name
+	team_names[team_number] = "Mothership [pick(possible_changeling_IDs)]" //TODO Ensure unique and actual alieny names
+	//Team Objective
+	var/datum/objective/experiment/team_objective = new
+	team_objective.team = team_number
+	team_objectives[team_number] = team_objective
+	//Team Members
+	if(antag_candidates.len >=2)
+		var/datum/mind/scientist = pick(antag_candidates)
+		antag_candidates -= scientist
+		var/datum/mind/agent = pick(antag_candidates)
+		antag_candidates -= agent
+
+		scientist.assigned_role = "MODE"
+		scientist.special_role = "Abductor"
+		log_game("[scientist.key] (ckey) has been selected as an abductor team [team_number] scientist.")
+
+		agent.assigned_role = "MODE"
+		agent.special_role = "Abductor"
+		log_game("[agent.key] (ckey) has been selected as an abductor team [team_number] agent.")
+
```

Comment:
Yep. Right now i placed four starting points. And it scales by config variable, defaults to 1 team per 15 people